lib/ffi-compiler/compile_task.rb

```diff
@@ -143,14 +148,25 @@ def define_task!
           index += 1
         end
 
-        ld = src_files.detect { |f| f =~ /\.cpp$/ } ? cxx : cc
+        ld = if src_files.detect { |f| f =~ /\.cpp$/ }
+               cxx
+             elsif src_files.detect { |f| f =~ /\.c$/ }
+               cc
+             else
```

Comment:
I think it's not good idea to use `else` but rather also compare extension like others and for else output warning about unknown extension.